test/factories/subnet.rb

```diff
@@ -9,15 +9,11 @@
     end
 
     trait :dhcp do
-      association :dhcp, :factory => :smart_proxy
+      association :dhcp, :factory => :dhcp_smart_proxy
     end
 
     trait :dns do
-      association :dns, :factory => :smart_proxy
-    end
-
-    trait :tftp do
```

Comment:
Not deleting trait, just removing duplicate. See the whole [factory](https://github.com/theforeman/foreman/blob/develop/test/factories/subnet.rb)